# HG changeset patch
# User Benjamin Jones <benjaminfjones@gmail.com>
# Date 1317408128 18000
# Node ID fb5d9ee43142c5fba6351abfbeefa6c9194c7396
# Parent  b1a099f40be3f25088294fdcb19612508d97b404
Trac 11885: added check that parent has attribute 'prec' before calling it

diff --git a/sage/libs/mpmath/utils.pyx b/sage/libs/mpmath/utils.pyx
--- a/sage/libs/mpmath/utils.pyx
+++ b/sage/libs/mpmath/utils.pyx
@@ -406,13 +406,21 @@
         sage: type(_)
         <type 'sage.rings.real_double.RealDoubleElement'>
 
+    Check that Trac 11885 is fixed::
+
+        sage: a.call(a.ei, 1.0r, parent=float)
+        1.8951178163559366
+
     """
     from mpmath import mp
     orig = mp.prec
     prec = kwargs.pop('prec', orig)
     parent = kwargs.pop('parent', None)
     if parent is not None:
-        prec = parent.prec()
+        try:
+            prec = parent.prec()
+        except AttributeError:
+            pass
     prec2 = prec + 20
     args = sage_to_mpmath(args, prec2)
     kwargs = sage_to_mpmath(kwargs, prec2)
